### PR TITLE
Update api-version of compute to 2020-06-01

### PR DESCRIPTION
**Pull request by ArcturusZhang**

### CODE REVIEW

File: azurerm/internal/services/compute/dedicated_host_group_data_source.go [hunk 1]
```diff
@@ -61,7 +62,7 @@ func dataSourceArmDedicatedHostGroupRead(d *schema.ResourceData, meta interface{
 	name := d.Get("name").(string)
 	resourceGroupName := d.Get("resource_group_name").(string)
 
-	resp, err := client.Get(ctx, resourceGroupName, name)
+	resp, err := client.Get(ctx, resourceGroupName, name, compute.InstanceView)
```
Comment (tombuildsstuff) [inline azurerm/internal/services/compute/dedicated_host_group_data_source.go:65]: `InstanceView` returns a different response from the API - presumably this wants to be an empty string?

File: azurerm/internal/services/compute/dedicated_host_group_resource.go [hunk 2]
```diff
@@ -75,7 +75,7 @@ func resourceArmDedicatedHostGroupCreate(d *schema.ResourceData, meta interface{
 	resourceGroupName := d.Get("resource_group_name").(string)
 
 	if d.IsNewResource() {
-		existing, err := client.Get(ctx, resourceGroupName, name)
+		existing, err := client.Get(ctx, resourceGroupName, name, compute.InstanceView)
```
Comment (tombuildsstuff) [inline azurerm/internal/services/compute/dedicated_host_group_resource.go:78]: `InstanceView` returns a different response from the API - presumably this wants to be an empty string?

File: azurerm/internal/services/compute/dedicated_host_group_resource.go [hunk 3]
```diff
@@ -105,7 +105,7 @@ func resourceArmDedicatedHostGroupCreate(d *schema.ResourceData, meta interface{
 		return fmt.Errorf("Error creating Dedicated Host Group %q (Resource Group %q): %+v", name, resourceGroupName, err)
 	}
 
-	resp, err := client.Get(ctx, resourceGroupName, name)
+	resp, err := client.Get(ctx, resourceGroupName, name, compute.InstanceView)
```
Comment (tombuildsstuff) [inline azurerm/internal/services/compute/dedicated_host_group_resource.go:108]: `InstanceView` returns a different response from the API - presumably this wants to be an empty string?

File: azurerm/internal/services/compute/dedicated_host_group_resource.go [hunk 4]
```diff
@@ -129,7 +129,7 @@ func resourceArmDedicatedHostGroupRead(d *schema.ResourceData, meta interface{})
 	resourceGroupName := id.ResourceGroup
 	name := id.Path["hostGroups"]
 
-	resp, err := client.Get(ctx, resourceGroupName, name)
+	resp, err := client.Get(ctx, resourceGroupName, name, compute.InstanceView)
```
Comment (tombuildsstuff) [inline azurerm/internal/services/compute/dedicated_host_group_resource.go:132]: `InstanceView` returns a different response from the API - presumably this wants to be an empty string?

File: azurerm/internal/services/compute/dedicated_host_resource.go [hunk 5]
```diff
@@ -172,7 +172,7 @@ func resourceArmDedicatedHostRead(d *schema.ResourceData, meta interface{}) erro
 		return err
 	}
 
-	group, err := groupsClient.Get(ctx, id.ResourceGroup, id.HostGroup)
+	group, err := groupsClient.Get(ctx, id.ResourceGroup, id.HostGroup, compute.InstanceView)
```
Comment (tombuildsstuff) [inline azurerm/internal/services/compute/dedicated_host_resource.go:175]: `InstanceView` returns a different response from the API - presumably this wants to be an empty string?

File: azurerm/internal/services/compute/linux_virtual_machine_scale_set_resource.go [hunk 6]
```diff
@@ -323,18 +323,18 @@ func resourceArmLinuxVirtualMachineScaleSetCreate(d *schema.ResourceData, meta i
 	rollingUpgradePolicyRaw := d.Get("rolling_upgrade_policy").([]interface{})
 	rollingUpgradePolicy := ExpandVirtualMachineScaleSetRollingUpgradePolicy(rollingUpgradePolicyRaw)
 
-	if upgradeMode != compute.Manual && healthProbeId == "" {
+	if upgradeMode != compute.UpgradeModeAutomatic && healthProbeId == "" {
```
Comment (tombuildsstuff) [inline azurerm/internal/services/compute/linux_virtual_machine_scale_set_resource.go:326]:
```suggestion
	if upgradeMode != compute.UpgradeModeManual && healthProbeId == "" {
```

**Comment by tombuildsstuff** [comment]
@ArcturusZhang out of interest what needs the API version upgrade?

File: azurerm/internal/services/compute/dedicated_host_group_data_source.go (hunk 1 above)
Comment (ArcturusZhang) [reply azurerm/internal/services/compute/dedicated_host_group_data_source.go:65]: Oops, sure it does make sense. Will change it right away

File: azurerm/internal/services/compute/dedicated_host_group_resource.go (hunk 2 above)
Comment (ArcturusZhang) [reply azurerm/internal/services/compute/dedicated_host_group_resource.go:78]: Will do

File: azurerm/internal/services/compute/dedicated_host_group_resource.go (hunk 3 above)
Comment (ArcturusZhang) [reply azurerm/internal/services/compute/dedicated_host_group_resource.go:108]: Done

File: azurerm/internal/services/compute/dedicated_host_resource.go (hunk 5 above)
Comment (ArcturusZhang) [reply azurerm/internal/services/compute/dedicated_host_resource.go:175]: DOne

File: azurerm/internal/services/compute/linux_virtual_machine_scale_set_resource.go (hunk 6 above)
Comment (ArcturusZhang) [reply azurerm/internal/services/compute/linux_virtual_machine_scale_set_resource.go:326]: Oops, my mistake

**Comment by ArcturusZhang** [comment]
Hi @tombuildsstuff This is required for one customer feature request from BlackRock for the new encryption on disks, virtual machines and virtual machine scale sets.

**Comment by tombuildsstuff** [comment]
Blocked on https://github.com/Azure/azure-rest-api-specs/issues/10373

**Comment by tombuildsstuff** [comment]
@ArcturusZhang since this is blocked on https://github.com/Azure/azure-rest-api-specs/issues/10373 I'm going to close this PR for the moment - but once that's fixed we should be able to re-open this and take another look.

**Pull request closed without merging**